### PR TITLE
Added workflow for html documentation

### PR DESCRIPTION
**Pull request by coldav**
# Overview

Added a pull request workflow for testing documentation

# Reason for change

There is no current PR testing for building the documentation.

# Description of change

 This adds a workflow which will build it and create an artifact, but not deploy it which will require a different workflow.

# Checklist

* Read and follow the project [Code of Conduct](https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/CODE_OF_CONDUCT.md).
* Make sure the project builds successfully with your changes.
* Run relevant testing locally to avoid regressions.
* Run [clang-format-9](https://clang.llvm.org/docs/ClangFormat.html) on all modified code.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: Install apt package
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y doxygen spirv-tools virtualenv
```
Comment (frasercrmck) [inline .github/workflows/docs.yml:29]: We need spirv-tools to build the documentation?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Install prerequisites
+        run: python3 -m pip install -r doc/requirements.txt
```
Comment (frasercrmck) [inline .github/workflows/docs.yml:32]: Do we need `python3` here or will `setup-python` create `python = python3` for us? Just trying to reduce the number of places we're encoding the version.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Build Documentation
+        working-directory: ${{github.workspace}}
+        run: |
+          cmake -DCA_ENABLE_API=cl -DCMAKE_BUILD_TYPE=Debug -DCA_RUNTIME_COMPILER_ENABLED=OFF -DCA_CL_ENABLE_OFFLINE_KERNEL_TESTS=OFF -DOCL_EXTENSION_cl_khr_il_program=OFF -Bbuild_doc
+          cmake --build build_doc  --target doc_html
```
Comment (frasercrmck) [inline .github/workflows/docs.yml:38]: Double space before `--target`

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Simple workflow for building documentation
+name: Build documentation
+
+# On a pull request this will build the html documentation
+# and create an artifact. It does this by using the standard
```
Comment (frasercrmck) [inline .github/workflows/docs.yml:5]: nit: "standard" build target made it sound to me like a github thing. I think we could be clearer here.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (coldav) [reply .github/workflows/docs.yml:29]: it seems to be required for the cmake - I agree that it shouldn't be needed. We may want to detach the documentation from the build in the future.

**Review by frasercrmck — APPROVED** [review]

**Pull request merged**